javascript/401-code-challenges/graph/index.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+class Graph {
+  constructor() {
+    this.graph = new Map();
+  }
+
+  addNode(value) {
+    const node = new Node(value);
+    this.graph.set(node, []);
+    return node;
+  }
+
+  addEdge(node1, node2, weight = null) {
+    if (this.graph.has(node1) && this.graph.has(node2)) {
+      const edge = new Edge(node1, node2, weight);
+      this.graph.get(node1).push(edge);
+      this.graph.get(node2).push(edge);
+    }
+  }
```

Comment:
This would make an undirected graph. Just be aware that in a directed graph, only node1 gets an edge added to its neighbors list.